src/ui/NoNameFacet/NoNameFacetHeader/NoNameFacetHeaderButton.ts

```diff
@@ -0,0 +1,34 @@
+import { $$ } from '../../../utils/Dom';
+import { SVGDom } from '../../../utils/SVGDom';
+
+export interface INoNameFacetHeaderButtonOptions {
+  label: string;
+  action?: () => void;
+  shouldDisplay: boolean;
+  className: string;
+  iconSVG: string;
+  iconClassName: string;
+}
+
+export abstract class NoNameFacetHeaderButton {
```

Comment:
I would not use a "abstract" / "extend" pattern.
I'd go with something that use composition instead. 
In my experience it is easier to be more flexible and change in the future.

A " facet header button" could be made of : 
* An icon (class in itself), only in charge if displaying the SVG
* An action, which holds the "click" code.
* A label.

And then you could do something like

```
class NoNameFacetHeaderClear {
   const icon = new HeaderIcon('./thesvg')
   const label = new Label('erase')
   const action = new Action(()=> this.erase())

   // or no icon if we don't need one.
   // or no label if we dont need one.. etc.
   // versus now with hierarchy, you kinda "need" to have an icon, etc..

  this.appendStuffInDom(icon, label, .... )
}
```